File: e72/bac/macro/n_index_mom.py
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.patches import Ellipse
from matplotlib.patches import Rectangle

# -----------------------
# Fixed parameters
# -----------------------
n_aero = 1.115 
pmin, pmax = 250.0, 1500.0         # momentum range [MeV/c]
sep_p1, sep_p2 = 625.0, 970.0    # separation region [MeV/c]

ellipse_height = 0.06            # height in refractive index
ellipse_alpha = 0.25

# particle masses [MeV/c^2]
m_pi = 139.57039
m_K  = 493.677

def n_threshold(p, m):
    """
    Cherenkov threshold:
    n_thr(p) = 1/beta = sqrt(1 + (m/p)^2)
    p, m in MeV units
    """
    p = np.asarray(p)
    return np.sqrt(1.0 + (m/p)**2)

# momentum grid (avoid zero)
p = np.linspace(20.0, pmax, 3000)

# threshold curves
n_pi = n_threshold(p, m_pi)
n_K  = n_threshold(p, m_K)

# -----------------------
# Plot
# -----------------------
fig, ax = plt.subplots(figsize=(8.5, 5.2))

ax.plot(p, n_pi, lw=2, label=r'$\pi$ threshold')
ax.plot(p, n_K,  lw=2, label=r'$K$ threshold')

# BAC refractive index line
ax.axhline(
    n_aero,
    color='black',
    ls='--',
    lw=2.5,
    label='Silica aerogel (n = 1.115)'
)

# separation ellipse

rect_height = 0.01     # 충분히 얇음 (0.005~0.01 추천)

rect = Rectangle(
    (sep_p1, n_aero - rect_height / 2),
    width=sep_p2 - sep_p1,
    height=rect_height,
    facecolor='red',
    alpha=0.25,
    edgecolor='none'
)

ax.add_patch(rect)

# vertical guide lines
ax.axvline(sep_p1, color='black', ls=':', lw=2)
ax.axvline(sep_p2, color='black', ls=':', lw=2)

# axis & style
ax.set_xlim(pmin, pmax)
ax.set_ylim(1.0, 1.35) 
ax.set_xlabel(r'Momentum [MeV/$c$]')
ax.set_ylabel('Refractive index')

ax.grid(True, alpha=0.3)
ax.legend(frameon=False, loc='upper right')


plt.tight_layout()
plt.savefig("pi_k_threshold.pdf")
plt.show()

File: e72/bac/macro/n_index_mom.cc
void n_index_mom()
{
  
  const double n_aero = 1.115;
  const double pmin = 530.0;
  const double pmax = 1030.0;
  const double pcenter = 735.0;

  const double sep_p1 = 645.0;
  const double sep_p2 = 825.0;

  const double m_pi = 139.57039;
  const double m_K  = 493.677;

  auto n_threshold = [](double p, double m) {
    return std::sqrt(1.0 + (m/p)*(m/p));
  };

  const int N = 3000;
  TGraph* g_pi = new TGraph();
  TGraph* g_K  = new TGraph();

  for (int i = 0; i < N; ++i) {
    double p = 20.0 + (pmax - 20.0) * i / (N - 1.0);
    g_pi->SetPoint(i, p, n_threshold(p, m_pi));
    g_K ->SetPoint(i, p, n_threshold(p, m_K));
  }

  TCanvas* c = new TCanvas("c", "pi K threshold", 850, 520);
  c->SetMargin(0.12, 0.05, 0.12, 0.05);
  //c->SetGrid(1, 1);

  TH1D* frame = new TH1D("frame", "", 100, pmin, pmax);
  frame->SetMinimum(1.0);
  frame->SetMaximum(1.35);
  frame->GetXaxis()->SetTitle("Momentum [MeV/c]");
  frame->GetYaxis()->SetTitle("Refractive index");
  frame->Draw();

  // Threshold curves
  g_pi->SetLineColor(kBlue + 1);
  g_pi->SetLineWidth(3);
  g_pi->Draw("L SAME");

  g_K->SetLineColor(kOrange + 7);
  g_K->SetLineWidth(3);
  g_K->Draw("L SAME");

  // Aerogel refractive index line
  TLine* line_n = new TLine(pmin, n_aero, pmax, n_aero);
  line_n->SetLineColor(kBlack);
  line_n->SetLineStyle(2);
  line_n->SetLineWidth(3);
  line_n->Draw("SAME");

  

  // Red separation region
  double rect_height = 0.01;
  TBox* box = new TBox(sep_p1, n_aero - rect_height/2.0,
                       sep_p2, n_aero + rect_height/2.0);
  box->SetFillColorAlpha(kRed, 0.25);
  box->SetLineColor(0);
  box->Draw("SAME");

  // Vertical guide lines
  TLine* v1 = new TLine(sep_p1, 1.0, sep_p1, 1.35);
  v1->SetLineStyle(9);
  v1->SetLineWidth(2);
  v1->Draw("SAME");

  TLine* v2 = new TLine(sep_p2, 1.0, sep_p2, 1.35);
  v2->SetLineStyle(9);
  v2->SetLineWidth(2);
  v2->Draw("SAME");

  //Main mom line
  TLine* v3 = new TLine(pcenter, 1.0, pcenter, 1.35);
  //v2->SetLineStyle(9);
  v3->SetLineWidth(4);
  v3->SetLineColor(kRed);
  v3->Draw("SAME");

  TLatex latex;
  latex.SetTextSize(0.05);
  latex.SetTextColor(kRed);
  latex.SetTextFont(132);
  latex.DrawLatex(742, 1.305, "735 MeV/#it{c}");

  // Legend
  TLegend* leg = new TLegend(0.63, 0.78, 0.90, 0.92);
  leg->AddEntry(g_pi, "#pi threshold", "l");
  leg->AddEntry(g_K,  "K threshold", "l");
  leg->AddEntry(line_n, "Silica aerogel (n = 1.115)", "l");
  leg->Draw();

  c->SaveAs("pi_k_threshold.pdf");
}
